src/js/util/svg.js

```diff
@@ -71,10 +71,7 @@ define(function (require, exports) {
         _.forEach(filter, function (kind) {
             var iconID = "layer-";
 
-            // No svg for these
-            if (kind === "solid color" || kind === "gradient" || kind === "pattern") {
-                iconID = "tool-rectangle";
-            } else if (kind === "artboard") {
+            if (kind === "artboard") {
```

Comment:
Yep, it works! When adding layer items, we add "artboard" as its category instead of group (`_getLayerCategory()` in actions/search/layers. 

This svg function uses the category list which is why the right icon shows up, and when the filter for artboards is on, it compares its value to the items' categories, so it finds the layers that are artboards as well.